third-culture-software/bhima
# Refactored handling of 'Include Exhausted Lots' flag in Stock Lots registry

Based on the discussion in Issue https://github.com/IMA-WorldHealth/bhima/issues/4862 I refactored how the 'Include Exhausted Lots' flag is handled:

- Changed the flag to a 3-value radio with very clear options (and no "clear" link)
- Moved the flag into the defaults tab for the Stock Lots registry and the Inventory in Stock registry.
- Added StockFilterExhaustedController to simplify using this flag.  Note that it includes some flexibility for the names of the options.   I thought we might possibly want different wording for the Items in stock vs lots registry.  If not we can remove that flexibility and hardcode the language tokens in the controller to make it even easier to use.
- Fixed the filtering so that it is possible to list ONLY exhausted lots (like the expired flag).
- Renamed language token STOCK.INCLUDE_EMPTY_LOT to STOCK.INCLUDE_EMPTY_LOTS for readability
- Added an exhausted lot in the Bhima_test data to simplify testing

Note that the flag now accepts 3 options:
![image](https://user-images.githubusercontent.com/62145/92293671-1fae0180-eeda-11ea-821e-ce922b6f38a1.png)


**Testing:**

- Make sure to do yarn build:db
- Do the stock lots registry and observe the 3 options and test them.
- Try the same in the the inventory in stock listing too.
